### PR TITLE
feat(datasource): manage datasources by org_name

### PR DESCRIPTION
**Pull request by Nemental**
##### SUMMARY

Like https://github.com/ansible-collections/community.grafana/pull/318 and https://github.com/ansible-collections/community.grafana/pull/331, but for `grafana_datasource` :)
Instead of specifying `org_id`, you can use `org_name`...

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

**module:** grafana_datasource
##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: plugins/modules/grafana_datasource.py [hunk 1]
```diff
@@ -679,7 +687,9 @@ def __init__(self, module):
             self.headers["Authorization"] = "Bearer %s" % module.params['grafana_api_key']
         else:
             self.headers["Authorization"] = basic_auth_header(module.params['url_username'], module.params['url_password'])
-            self.switch_organisation(module.params['org_id'])
+            if module.params['org_name']:
+                module.params['org_id'] = self.organization_by_name(module.params['org_name'])
+            self.switch_organization(module.params['org_id'])
```
Comment (rndmh3ro) [inline plugins/modules/grafana_datasource.py:692]: I'm not a fan of re-declaring this variable. How about we define a new one and use the ternay-operator:

```suggestion
            org_id = self.organization_by_name(module.params['org_name']) if module.params['org_name'] else module.params['org_id']
            self.switch_organization(org_id)
```

File: plugins/modules/grafana_datasource.py [hunk 2]
```diff
@@ -701,10 +711,19 @@ def _send_request(self, url, data=None, headers=None, method="GET"):
...
+    def switch_organization(self, org_id):
         url = "/api/user/using/%d" % org_id
         response = self._send_request(url, headers=self.headers, method='POST')
 
+    def organization_by_name(self, org_name):
+        url = "/api/user/orgs"
+        organizations = self._send_request(url, headers=self.headers, method='GET')
+        for org in organizations:
+            if org['name'] == org_name:
+                return org['orgId']
```
Comment (rndmh3ro) [inline plugins/modules/grafana_datasource.py:723]: We could use next + list comprehension here, making it shorter and faster:
https://stackoverflow.com/questions/9014058/creating-a-python-list-comprehension-with-an-if-and-break

```suggestion
            org = next((org for org in organizations if org['name'] == org_name))
            if org:
                return org['orgId']
```

**Review by rndmh3ro — COMMENTED** [review]
LGTM!

I left some possible improvements.

**Comment by Nemental** [comment]
@rndmh3ro great improvements! Thank you 🚀
Didn't know about the list comprehension, nice 👍

**Review by rndmh3ro — APPROVED** [review]

**Pull request merged**